### PR TITLE
Feature/coverage

### LINKED ISSUE #588: Fix coverage run to not hang in CI

### PR DESCRIPTION
**Pull request by ruseinov**
Fixes #588

### CODE REVIEW

**Comment by ruseinov** [comment]
This seems to have fixed itself after tendermint and client updates.

**Comment by ethanfrey** [comment]
Awesome!
I am going to try this out now

File: .travis.yml [hunk 1]
```diff
@@ -38,12 +38,10 @@ install:
...
 - make lint;
 - make test;
-# make cover seems to hang on client tests, and we cannot debug them 
-# TODO: fix issue and re-enable
-# - if [[ "$TRAVIS_GO_VERSION" == "$MAIN_GO_VERSION" ]]; then
-#     make cover;
-#     ./coverage/upload.sh;
-#   fi;
+- if [[ "$TRAVIS_GO_VERSION" == "$MAIN_GO_VERSION" ]]; then
+     make cover;
```
Comment (ethanfrey) [inline .travis.yml]: :+1:

File: cmd/bnsd/scenarios/main_test.go [hunk 2]
```diff
@@ -91,8 +92,17 @@ func TestMain(m *testing.M) {
 	// run the app inside a tendermint instance
 	node = rpctest.StartTendermint(app)
 	bnsClient = client.NewClient(client.NewLocalConnection(node))
-	time.Sleep(startupDelay) // wait for chain
-	code := m.Run()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
```
Comment (ethanfrey) [inline cmd/bnsd/scenarios/main_test.go]: Nice job adding the adaptive startup time here

File: cmd/bnsd/scenarios/main_test.go [hunk 3]
```diff
@@ -91,8 +92,17 @@ func TestMain(m *testing.M) {
...
+	defer cancel()
+	_, err = weaveClient.NewClient(weaveClient.NewLocalConnection(node)).WaitForNextBlock(ctx)
```
Comment (ethanfrey) [inline cmd/bnsd/scenarios/main_test.go]: Question: this usage seems clean enough, but a little verbose. Do you have any concrete suggestions to make this simpler? Maybe combine NewClient and NewLocalConnection into NewLocalClient??

Comment (ruseinov) [reply cmd/bnsd/scenarios/main_test.go]: Yes, I looked at that and I was even thinking to have a helper of some sort in client lib to reuse here too. I'll create a ticket for that

**Pull request merged**